polkadot/node/core/prospective-parachains/src/fragment_chain/mod.rs

```diff
@@ -679,28 +782,28 @@ impl FragmentChain {
 	// Stops when the ancestors are all used or when a node in the chain is not present in the
 	// ancestor set. Returns the index in the chain were the search stopped.
 	fn find_ancestor_path(&self, mut ancestors: Ancestors) -> usize {
-		if self.chain.is_empty() {
+		if self.best_chain.chain.is_empty() {
 			return 0;
 		}
 
-		for (index, candidate) in self.chain.iter().enumerate() {
+		for (index, candidate) in self.best_chain.chain.iter().enumerate() {
 			if !ancestors.remove(&candidate.candidate_hash) {
```

Comment:
Due to the way the provisioner is coded, this can happen if a candidate times out availability.

The provisioner does not do any ordering on the ancestors (which would increase complexity, as it doesn't have the right data yet and the candidates are not occupying the cores in order). So if a candidates times out at this block, it will not add it to the ancestors set (but it will add the descendants of the timed out candidate).

That's why we stop when we find a hole